src/plugins/intel_gpu/src/plugin/compiled_model.cpp

```diff
@@ -187,6 +187,9 @@ std::shared_ptr<ov::IAsyncInferRequest> CompiledModel::create_infer_request() co
 //     [ ov::Node::Input/ ov::Node::Output ]
 //     [ ov::intel_gpu::Graph ]
 void CompiledModel::export_model(std::ostream& model) const {
+    if (m_config.get_property(ov::cache_mode) == ov::CacheMode::OPTIMIZE_SIZE)
```

Comment:
We don't have robust method to check whether we are inside "model cache" mode.

We could potentially check that `std::stream` is empty or not, because in case of "model cache" we typically write compiled blob header before calling CompiledModel::export_model. But it will not work in HETERO case, when multiple sub-graph are written to the same stream and GPU is not called first..